yassersharabati/main.py

```diff
@@ -0,0 +1,50 @@
+from fastapi import FastAPI, HTTPException
+from uuid import uuid4
+from typing import List
+from task import *
+
+app = FastAPI()
+
+
+tasks = []
+
+@app.post("/tasks/", response_model=Task)
+def create_task(task: Task):
+    task.task_id = str(uuid4()) 
+    tasks.append(task)
+    return task
+
+
+@app.put("/tasks/{task_id}/", response_model=Task)
+def update_task(task_id: str, updated_task: Task):
```

Comment:
Here the type of `updated_task` should have the same properties as the `Task` but all of them are optional to validate them in a better way